### PR TITLE
fix pg_Stat_activity display and Improve output warning when pg_resgroup_move_query

### PR DESCRIPTION
**Pull request by yanwr1**
For unassigned and bypassed query, still show resgroup information as before, but append "*" to end of the group name to imply it's a bypass query.

Improve the output warning when execute pg_resgroup_move_query for process in IDLE state.

## Here are some reminders before you submit the pull request
- [ ] Add tests for the change
- [ ] Document changes
- [ ] Communicate in the mailing list if needed
- [ ] Pass `make installcheck`
- [ ] Review a PR in return to support the community

### CODE REVIEW

File: src/backend/postmaster/pgstat.c [hunk 1]
```diff
@@ -3510,7 +3510,7 @@ pgstat_report_xact_timestamp(TimestampTz tstamp)
  * Report the timestamp of transaction start queueing on the resource group.
  */
 void
-pgstat_report_resgroup(Oid groupid)
+pgstat_report_resgroup(Oid groupid, bool bypass)
```
Comment (kainwen) [inline src/backend/postmaster/pgstat.c:3513]: ABI is locked since beta5. Can we find another way?

Comment (yanwr1) [reply src/backend/postmaster/pgstat.c]: changed to use bypassedGroup to decide whether query is in bypass mode

File: src/backend/postmaster/pgstat.c [hunk 2]
```diff
@@ -3525,6 +3526,8 @@ pgstat_report_resgroup(Oid groupid)
 	beentry->st_changecount++;
 
 	beentry->st_rsgid = groupid;
+	if (ResGroupIsBypassed())
+		beentry->st_bypass_rsg = true;
```
Comment (kainwen) [inline src/backend/postmaster/pgstat.c:3530]: What about `beentry->st_bypass_rsg = ResGroupIsBypassed();`?

File: src/backend/utils/resgroup/resgroup_helper.c [hunk 3]
```diff
@@ -497,7 +497,7 @@ pg_resgroup_move_query(PG_FUNCTION_ARGS)
 		if (currentGroupId == InvalidOid)
 			ereport(ERROR,
 					(errcode(ERRCODE_UNDEFINED_OBJECT),
-							(errmsg("process %d is in IDLE state", pid))));
```
Comment (kainwen) [inline src/backend/utils/resgroup/resgroup_helper.c:500]: What about not throw error and just a warning?

Comment (yanwr1) [reply src/backend/utils/resgroup/resgroup_helper.c:500]: the doc workload_mgmt_resgroups.html.md says: You can move only an active or running query to a new resource group. You cannot move a queued or pending query that is in an idle state due to concurrency or memory limits.
move query in idle state is not supported.

**Comment by yanwr1** [comment]
current performance:
```
postgres=# select * from pg_stat_activity;
 datid | datname  |   pid   | sess_id | usesysid | usename | application_name | client_addr | client_hostname | client_port |  
       backend_start         |          xact_start           |          query_start          |         state_change          | 
wait_event_type |     wait_event      |        state        | backend_xid | backend_xmin |                  query              
    |         backend_type         | rsgid |       rsgname       
-------+----------+---------+---------+----------+---------+------------------+-------------+-----------------+-------------+--
-----------------------------+-------------------------------+-------------------------------+-------------------------------+-
----------------+---------------------+---------------------+-------------+--------------+-------------------------------------
----+------------------------------+-------+---------------------
       |          | 3222650 |       1 |          |         |                  |             |                 |             | 2
023-09-13 06:43:13.90264+00  |                               |                               |                               | 
Activity        | AutoVacuumMain      |                     |             |              |                                     
    | autovacuum launcher          |     0 | unknown
 13720 | postgres | 3222652 |       3 |       10 | gpadmin |                  |             |                 |             | 2
023-09-13 06:43:13.903693+00 |                               |                               |                               | 
IPC             | DtxRecovery         |                     |             |              |                                     
    | dtx recovery process         |     0 | unknown
       |          | 3222672 |       4 |       10 | gpadmin |                  |             |                 |             | 2
023-09-13 06:43:14.131085+00 |                               |                               |                               | 
Activity        | LogicalLauncherMain |                     |             |              |                                     
    | logical replication launcher |     0 | unknown
       |          | 3222865 |      -1 |       10 | gpadmin | gp_walreceiver   | 127.0.0.1   |                 |       11119 | 2
023-09-13 06:43:15.292562+00 |                               | 2023-09-13 06:43:15.294191+00 | 2023-09-13 06:43:15.294205+00 | 
Activity        | WalSenderMain       | active              |             |              | START_REPLICATION 0/14000000 TIMELIN
E 1 | walsender                    |     0 | unknown
 13720 | postgres | 3223048 |       7 |       10 | gpadmin | psql             |             |                 |          -1 | 2
023-09-13 06:43:22.081747+00 | 2023-09-13 06:46:32.489813+00 | 2023-09-13 06:46:32.489813+00 | 2023-09-13 06:46:32.489817+00 | 
                |                     | active              |             |         1695 | select * from pg_stat_activity;     
    | client backend               |  6438 | admin_group<bypass>
 13720 | postgres | 3223083 |       8 |       10 | gpadmin | psql             |             |                 |          -1 | 2
023-09-13 06:43:38.224377+00 | 2023-09-13 06:45:26.144025+00 | 2023-09-13 06:45:26.144025+00 | 2023-09-13 06:45:26.144029+00 | 
                |                     | active              |             |         1695 | select * from pg_class limit 1;     
    | client backend               |  6438 | admin_group<bypass>
 13720 | postgres | 3223393 |      15 |       10 | gpadmin | psql             |             |                 |          -1 | 2
023-09-13 06:46:23.873324+00 | 2023-09-13 06:46:26.274464+00 | 2023-09-13 06:46:26.274464+00 | 2023-09-13 06:46:26.287944+00 | 
Client          | ClientRead          | idle in transaction |             |              | begin;                              
    | client backend               |  6438 | admin_group
       |          | 3222648 |      -1 |          |         |                  |             |                 |             | 2
023-09-13 06:43:13.902123+00 |                               |                               |                               | 
Activity        | BgWriterHibernate   |                     |             |              |                                     
    | background writer            |     0 | unknown
       |          | 3222647 |      -1 |          |         |                  |             |                 |             | 2
023-09-13 06:43:13.901902+00 |                               |                               |                               | 
Activity        | CheckpointerMain    |                     |             |              |                                     
    | checkpointer                 |     0 | unknown
       |          | 3222649 |      -1 |          |         |                  |             |                 |             | 2

postgres=# select pg_resgroup_move_query(3223083,'default_group');
ERROR:  cannot move a bypassed or unassigned query
postgres=# select pg_resgroup_move_query(3223083,'default_group');
ERROR:  process 3223083 is in IDLE state, cannot move a queued or pending query in idle state

**Review by kainwen — APPROVED** [review]
LGTM.

**Comment by yanwr1** [comment]
dev pipeline is all green: https://dev.ci.gpdb.pivotal.io/teams/main/pipelines/resgroup

**Pull request merged**